src/components/map/FeatureLoadService.js

```diff
@@ -0,0 +1,64 @@
+
+goog.provide('ga_featureload_service');
+
+goog.require('ga_map_service');
+goog.require('ga_styles_service');
+
+(function() {
+  var module = angular.module('ga_featureload_service', [
+    'ga_map_service',
+    'ga_styles_service'
+  ]);
+
+  /*
+   *This service checks if features are loaded.
+   */
+
+  module.provider('gaFeatureLoadManager', function() {
+    this.$get = function($rootScope, $q, $http, gaDefinePropertiesForLayer,
+        gaStyleFactory, gaLayerFilters, gaMapUtils, gaLayers) {
+
+      var promises = {};
+      var scope = $rootScope.$new();
+
+      var FeatureLoadManager = function() {
+        var map;
+
+        // watches layers for changes, returns loaded features in promises.
+        var createWatcher = function(bodId) {
+          scope.layers = map.getLayers().getArray();
+          scope.layerFilter = gaLayerFilters.geojson;
+          scope.$watchCollection('layers | filter:layerFilter',
+              function(layers) {
+                var layerFeatures;
+                angular.forEach(layers, function(flayer) {
+                  flayer.getSource().on('addfeature', function(event) {
```

Comment:
The `addFeature` event is sent for each  feature added, but you only want to execute code once, when the GeoJSON is loaded, that's why the use of promises created in the `gaLayers` service makes more sense.

Reply:
That's true. We still have all features on first call though.

Unfortunately, ol doesn't offer a 'featureload' event - which would make the most sense, I think.